File: ToolDemo/app/src/main/java/com/lilanz/tooldemo/utils/NumberUtil.java
package com.lilanz.tooldemo.utils;


import android.support.annotation.NonNull;

import java.util.regex.Pattern;

public class NumberUtil {

    /**
     * @param hexStr 屏幕端显示的16进制数据
     * @return 屏幕端显示的10进制数据
     */
    public static int hex2Int(@NonNull String hexStr) {
        int count = 0;
        int num = 0;
        for (char c : hexStr.toCharArray()) {
            switch (c) {
                case 'a':
                case 'A':
                    num = 10;
                    break;
                case 'b':
                case 'B':
                    num = 11;
                    break;
                case 'c':
                case 'C':
                    num = 12;
                    break;
                case 'd':
                case 'D':
                    num = 13;
                    break;
                case 'e':
                case 'E':
                    num = 14;
                    break;
                case 'f':
                case 'F':
                    num = 15;
                    break;
                default:
                    num = c - '0';
                    break;
            }
            count = count * 16 + num;
        }
        return count;
    }

}

File: KotlinTool/app/src/main/java/com/lilanz/kotlintool/utils/NumberUtil.kt
package com.lilanz.kotlintool.utils

import java.lang.Exception

class NumberUtil {
    companion object {
        /**
         * 判断字符串是不是整数
         */
        fun isInt(str: String): Boolean {
            return try {
                str.toInt()
                true
            } catch (e: Exception) {
                false
            }
        }

        /**
         * 判断字符串是不是整数
         */
        fun isNumber(str: String): Boolean {
            return try {
                str.toDouble()
                true
            } catch (e: Exception) {
                false
            }
        }

        /**
         * 十六进制字符串转十进制数值
         */
        fun hex2Int(str: String): Int {
            if (str == "") {
                return 0
            }

            var count = 0;
            var num = 0
            for (c in str.toCharArray()) {
                num = when (c) {
                    'a', 'A' -> 10
                    'b', 'B' -> 11
                    'c', 'C' -> 12
                    'd', 'D' -> 13
                    'e', 'E' -> 14
                    'f', 'F' -> 15
                    else -> c - '0'
                }
                count = count * 16 + num
            }

            return count
        }
    }
}